File: Array/01.TwoSum/TwoSum.java

// （时间O（n），空间O（n），通俗易懂）

import java.util.HashMap;

public class Solution {

    public int[] twoSum(int[] nums, int target) {

        HashMap<Integer,Integer> hashMap = new HashMap<>();

        for (int i = 0; i  < nums.length; i++){

            int key = target - nums[i];

            if (hashMap.containsKey(key)){

                return new int[]{hashMap.get(key), i};

            }

            hashMap.put(nums[i],i);

        }

        throw new IllegalArgumentException("No two sum solution");

    }

}

File: Array/01.TwoSum/TwoSum.cpp

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> res;
        map<int, int> hashMap;
        for(int i=0; i<nums.size(); ++i) {
            int y = target - nums[i];
            auto iter = hashMap.find(y);
            if(iter != hashMap.end()) {
                res = {hashMap[y], i};
                break;
            }
            hashMap[nums[i]] = i;
        }
        return res;
    }
};